data/mods/m4av7/abilities.ts

```diff
@@ -0,0 +1,1493 @@
+				if (target.hasAbility('arenarock')) {
+					this.field.terrainData.source = target;
+					return;
+				}
+			}
+			this.field.clearTerrain();
+		},
+		name: "Arena Rock",
+		rating: 4.5,
+		num: -1004,
+	},
+	sharpstriker: {
+		desc: "This Pokémon's ballistic moves have their power multiplied by 1.5. Ballistic moves include Bullet Seed, Octazooka, Barrage, Rock Wrecker, Zap Cannon, Acid Spray, Aura Sphere, Focus Blast, and all moves with Ball or Bomb in their name.",
+		shortDesc: "This Pokémon's ballistic moves have 1.5x power (Shadow Ball, Sludge Bomb, Focus Blast, etc).",
+		onBasePowerPriority: 19,
+		onBasePower(basePower, attacker, defender, move) {
+			if (move.flags['bullet']) {
+				return this.chainModify(1.5);
+			}
+		},
+		name: "Sharp Striker",
+		rating: 3,
+		num: -1005,
+	},
+	coldsweat: {
+		desc: "On switch-in, this Pokémon summons hail. It changes the current weather to rain whenever any opposing Pokémon has an attack that is super effective on this Pokémon or an OHKO move. Counter, Metal Burst, and Mirror Coat count as attacking moves of their respective types, Hidden Power counts as its determined type, and Judgment, Multi-Attack, Natural Gift, Revelation Dance, Techno Blast, and Weather Ball are considered Normal-type moves.",
+		shortDesc: "Summons hail on switch-in. Changes weather to rain if the foe has a supereffective or OHKO move.",
+		onStart(source) {
+			this.field.setWeather('hail');
+			for (const target of source.side.foe.active) {
+				if (!target || target.fainted) continue;
+				for (const moveSlot of target.moveSlots) {
+					const move = this.dex.getMove(moveSlot.move);
+					if (move.category === 'Status') continue;
+					const moveType = move.id === 'hiddenpower' ? target.hpType : move.type;
+					if (
+						this.dex.getImmunity(moveType, source) && this.dex.getEffectiveness(moveType, source) > 0 ||
+						move.ohko
+					) {
+						this.field.setWeather('raindance');
+						return;
+					}
+				}
+			}
+		},
+		onAnySwitchIn(pokemon) {
+			const source = this.effectData.target;
+			if (pokemon === source) return;
+			for (const target of source.side.foe.active) {
+				if (!target || target.fainted) continue;
+				for (const moveSlot of target.moveSlots) {
+					const move = this.dex.getMove(moveSlot.move);
+					if (move.category === 'Status') continue;
+					const moveType = move.id === 'hiddenpower' ? target.hpType : move.type;
+					if (
+						this.dex.getImmunity(moveType, source) && this.dex.getEffectiveness(moveType, source) > 0 ||
+						move.ohko
+					) {
+						this.field.setWeather('raindance', source);
+						return;
+					}
+				}
+			}
+		},
+		name: "Cold Sweat",
+		rating: 4,
+		num: -1006,
+	},
+	trashcompactor: {
+		desc: "This Pokémon is immune to all entry hazards. If it lands on any type of entry hazard, it clears the hazard and Stockpiles 1.",
+		shortDesc: "Hazard immunity. Clears hazards, Stockpiles 1 if switched in on them.",
+		onAfterMega(pokemon) {
```

Comment:
Actually, I can't tell if `AfterMega` is a custom event or just not in the `AbilityData` definition. Feel free to add it (I guess to `EventMethods`?

Reply:
Oh yeah, it's a real event. Just add it to `EventMethods` in `dex-conditions` @KrisXV